c++/src/kj/compat/http.c++

```diff
@@ -4345,16 +4753,25 @@ public:
 
     kj::Maybe<kj::String> offeredExtensions;
     kj::Maybe<CompressionParameters> clientOffer;
-    if (settings.enableWebSocketCompression) {
+    kj::Vector<CompressionParameters> extensions;
+    auto compressionMode = settings.webSocketCompressionMode;
+
+    if (compressionMode == HttpClientSettings::MANUAL_COMPRESSION) {
       KJ_IF_MAYBE(value, headers.get(HttpHeaderId::SEC_WEBSOCKET_EXTENSIONS)) {
         // Strip all `Sec-WebSocket-Extensions` except for `permessage-deflate`.
-        auto extensions = _::findValidExtensionOffers(*value);
-        if (extensions.size() > 0) {
-          clientOffer = extensions.front();
-          connectionHeaders[HttpHeaders::BuiltinIndices::SEC_WEBSOCKET_EXTENSIONS] =
-              offeredExtensions.emplace(_::generateExtensionRequest(extensions.asPtr()));
-        }
+        extensions = _::findValidExtensionOffers(*value);
       }
+    } else if (compressionMode == HttpClientSettings::AUTOMATIC_COMPRESSION) {
+      // If AUTOMATIC_COMPRESSION is enabled, we send `Sec-WebSocket-Extensions: permessage-deflate`
+      // to the server and ignore the `headers` provided by the caller.
+      constexpr auto default_config = "permessage-deflate";
```

Comment:
Should use camel case: `defaultConfig`

However, do we really want to initialize the config by parsing a string? Couldn't you do `extensions.add(CompressionParameters())` instead and avoid the parsing?

Reply:
I think my assumption was if someone wanted to use `AUTOMATIC_COMPRESSION` but also add parameters then it'd be easier to just modify the string, but I guess in that case it would make more sense to use `MANUAL_COMPRESSION`.

Reply:
https://github.com/capnproto/capnproto/pull/1505/commits/c350aaf975f89d5e16aaa613c3535ce5d21112d1